### ISSUE TITLE
Fails to detect bad file_mode when notify

### ISSUE THREAD

**Issue by maxb** [puppetlabs/puppet-lint#150]
A `file_mode` warning which should be generated, is not.

Given this `sample.pp`:
```
service {
  'zero':
}

file {
  '/one':
    mode   => '755',
    notify => Service['zero'];

  '/two':
    mode => '755';
}
```
and
```
puppet-lint --config /dev/null sample.pp
```
one would expect `file_mode` warnings about both `mode` arguments.

But actually, only the first is warned about:
```
WARNING: mode should be represented as a 4 digit octal value or symbolic mode on line 7 (check: file_mode)
```

Something about the line
```
    notify => Service['zero'];
```
causes the following instance to not generate a lint warning.

## Environment
```
$ puppet-lint --version
puppet-lint 4.2.0
```
